src/geometric.js

```diff
@@ -0,0 +1,11 @@
+export default function(p) {
+	if (arguments.length != 1) {
+		throw new SyntaxError("geometric(probability) must be called with only the probability parameter.");
+	}
+	if(p <= 0 || p > 1) {
+		throw new SyntaxError("geometric(probability) should be called with 0 < probability <= 1.");
+	}
```

Comment:
Here I think it does make sense to do some enforcement in addition to coercion, since if `p` is greater than 1 this random number generator will always return NaN. So my first thought is to say:

```js
if ((p = +p) <= 0 || p > 1) throw new RangeError("invalid p");
```

(Alternatively we could clamp the value, `p = Math.max(0, Math.min(1, p))`, but I think a RangeError is fine…)

However, I am wary of the strictly-greater-than-zero test, since it seems like due to floating point precision the behavior of this random number generator breaks with very small positive values such as Number.MIN_VALUE.

So it would be more accurate to test that `1 - p < 1` than to test `p > 0`. For example:

```js
p = 0.00000000000000001
p > 0 // true
1 - p // 1
1 - p < 1 // false
```

For that matter it’s slightly faster if we set `p = 1 - p` since we avoid an extra subtraction when generating the random number.

Here is what I would write:

```js
export default function(p) {
  if ((p = 1 - p) < 0 || p >= 1) throw new RangeError("invalid p");
  return function() {
    return 1 + Math.floor(Math.log(Math.random()) / Math.log(p));
  };
}
```